apps/desktop/src/components/Editor/decorations.ts

```diff
@@ -0,0 +1,272 @@
+    // 3. Combine them.
+
+    // Actually, to keep it simple and performant, we can just use one builder if we are careful,
+    // but since we iterate syntaxTree (which yields nodes in order), injecting wikilinks "in between" is hard.
+    //
+    // Better approach: Return a DecorationSet that combines multiple RangeSets if needed, or better yet,
+    // since we want to output a single DecorationSet, we can just create a separate set for wikilinks
+    // and let CodeMirror handle multiple sets in the ViewPlugin?
+    // No, ViewPlugin expects a single DecorationSet usually, but we can return `Decoration.set([set1, set2])`.
+    //
+    // Let's refactor `buildDecorations` to return an array of sets or just one merged set.
+  }
+
+  // Wait, RangeSetBuilder MUST be added to in order.
+  // The syntaxTree iteration happens in order.
+  // The wikiLinks are in order.
+  // But they interleave.
+  //
+  // A cleaner way is to collect ALL decorations into an array first, sort them by `from`, and then build.
+
+  return builder.finish();
+}
+
+/**
+ * We need to handle the interleaving of decorations from different sources (Syntax Tree vs Manual Regex).
+ * Since RangeSetBuilder requires ordered addition, we will collect all decoration requests into a buffer,
+ * sort them, and then build the set.
+ */
+function buildCombinedDecorations(view: EditorView): DecorationSet {
+    const decorations: Array<{from: number, to: number, deco: Decoration}> = [];
+
+    // 1. Frontmatter
+    const frontmatter = detectFrontmatter(view);
+    if (frontmatter) {
+         if (frontmatter.to >= view.viewport.from && frontmatter.from <= view.viewport.to) {
+            decorations.push({
+                from: frontmatter.from,
+                to: frontmatter.to,
+                deco: Decoration.mark({ class: 'cm-frontmatter' })
+            });
+         }
+    }
+
+    // 2. Syntax Tree (Standard Markdown)
+    for (const { from, to } of view.visibleRanges) {
+        syntaxTree(view.state).iterate({
+            from,
+            to,
+            enter(node) {
+                const { type, from, to } = node;
+
+                if (type.name === 'StrongEmphasis') {
+                    decorations.push({ from, to, deco: Decoration.mark({ class: 'cm-strong' }) });
+                } else if (type.name === 'Emphasis') {
+                     decorations.push({ from, to, deco: Decoration.mark({ class: 'cm-emphasis' }) });
+                } else if (type.name === 'InlineCode') {
+                     decorations.push({ from, to, deco: Decoration.mark({ class: 'cm-inline-code' }) });
+                } else if (type.name.startsWith('ATXHeading')) {
+                     decorations.push({ from, to, deco: Decoration.mark({ class: `cm-heading cm-heading-${type.name.slice(-1)}` }) });
+                } else if (type.name === 'Link') {
+                     decorations.push({ from, to, deco: Decoration.mark({ class: 'cm-link' }) });
+                }
```

Comment:
Can you review all the commands that currently exist and make sure the editor knows how to present them inline with the formatting visible.  Eg.Some of the commands, like Strikethrough do not work currently.

<img width="1032" height="63" alt="Image" src="https://github.com/user-attachments/assets/d399db49-29e2-4978-bda8-6566a670b98a" />